src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -26490,73 +26776,1152 @@ void Validate(ModuleSymbol m)
-        {
-            foreach (var options in new[] { TestOptions.DebugExe, TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All) })
-            {
-                var compilation1 = CreateCompilation(source1 + source2, options: options,
-                                                     parseOptions: TestOptions.Regular,
-                                                     targetFramework: TargetFramework.NetStandardLatest);
-                Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
-                compilation1.VerifyDiagnostics();
+            var compilation2 = CreateCompilation(source2, new[] { compilation1.ToMetadataReference() }, options: TestOptions.DebugDll,
+                                                    parseOptions: TestOptions.Regular,
+                                                    targetFramework: TargetFramework.NetStandardLatest);
 
-                ValidatePropertyImplementationInDerived_01(compilation1.SourceModule);
+            compilation2.VerifyDiagnostics(
+                // (2,7): warning CS8645: 'I1<string?>' is already listed in the interface list on type 'Test1' with different nullability of reference types.
+                // class Test1 : I2, I3
+                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "Test1").WithArguments("I1<string?>", "Test1").WithLocation(2, 7),
+                // (2,15): error CS8705: Interface member 'I1<string>.M1()' does not have a most specific implementation. Neither 'I2.I1<string>.M1()', nor 'I3.I1<string?>.M1()' are most specific.
+                // class Test1 : I2, I3
+                Diagnostic(ErrorCode.ERR_MostSpecificImplementationIsNotFound, "I2").WithArguments("I1<string>.M1()", "I2.I1<string>.M1()", "I3.I1<string?>.M1()").WithLocation(2, 15),
+                // (2,19): error CS8705: Interface member 'I1<string?>.M1()' does not have a most specific implementation. Neither 'I2.I1<string>.M1()', nor 'I3.I1<string?>.M1()' are most specific.
+                // class Test1 : I2, I3
+                Diagnostic(ErrorCode.ERR_MostSpecificImplementationIsNotFound, "I3").WithArguments("I1<string?>.M1()", "I2.I1<string>.M1()", "I3.I1<string?>.M1()").WithLocation(2, 19)
+                );
+
+            var test1 = compilation2.GetTypeByMetadataName("Test1");
+
+            Assert.Equal(new[] { "I2", "I1<System.String>", "I3", "I1<System.String?>" },
+                         test1.AllInterfacesNoUseSiteDiagnostics.Select(i => i.ToTestDisplayString()));
+
+            Assert.Null(test1.FindImplementationForInterfaceMember(test1.AllInterfacesNoUseSiteDiagnostics[1].GetMember("M1")));
+            Assert.Null(test1.FindImplementationForInterfaceMember(test1.AllInterfacesNoUseSiteDiagnostics[3].GetMember("M1")));
+
+            var compilation3 = CreateCompilation(source2, new[] { compilation1.EmitToImageReference() }, options: TestOptions.DebugDll,
```

Comment:
Consider making this test a `Theory` with a flag to control whether or not the reference to the first compilation is image or metadata, to make it clearer that these should have the same behavior.

Reply:
Same comment for the similar tests below.

---
In reply to: [262234539](https://github.com/dotnet/roslyn/pull/33778#discussion_r262234539) [](ancestors = 262234539)

Reply:
> to make it clearer that these should have the same behavior.

There is actually a slight difference in behavior. Nullability is not preserved for an explicit implementation ```I3.I1<string?>.M1``` in metadata.

---
In reply to: [262234970](https://github.com/dotnet/roslyn/pull/33778#discussion_r262234970) [](ancestors = 262234970,262234539)